tests/functions.sh

```diff
@@ -229,19 +236,34 @@ function ssh_clean_connection {
     ssh_run $id2 "rm /tmp/test" &> /dev/null || true
     ssh_run $id1 "killall ncat" &> /dev/null || true
     ssh_run $id2 "killall ncat" &> /dev/null || true
+    ssh_run $id2 "killall sctp_test" &> /dev/null || true
 }
 
 function ssh_connection_test_file {
+    id=$1
+    protocol=$2
     timeout=50
     it=0
-    while [ $it -ne $timeout ]; do
-	ssh_run $1 [ -s "/tmp/test" ]
-	if [ "$?" == "0" ]; then
-	    return 0
-	fi
-	sleep 0.1
-	it=$( expr $it + 1 )
-    done
+
+    if [ "$protocol" == "sctp" ]; then
+	while [ $it -ne $timeout ]; do
+	    ret=$(ssh_run $id "sed -n '$=' /tmp/test")
+	    if [ "$ret" -gt "20" ]; then
+		return 0
+	    fi
+	    sleep 0.1
+	    it=$( expr $it + 1 )
+	done
+    else
+	while [ $it -ne $timeout ]; do
+	    ssh_run $1 [ -s "/tmp/test" ]
```

Comment:
dont use directly $1, set variable at the beginning of the function like `myparam=$1`